unyt/mpl_interface.py

```diff
@@ -98,11 +99,39 @@ def convert(value, unit, axis):
             Raises
             ------
 
-            ConversionError if unit does not have the same dimensions as value
+            ConversionError if unit does not have the same dimensions as value or
+            if we don't know how to convert value.
             """
             if isinstance(unit, str) or isinstance(unit, Unit):
                 unit = (unit,)
-            return value.to(*unit)
+            if isinstance(value, (unyt_array, unyt_quantity)):
+                converted_value = value.to(*unit)
+            elif isinstance(value, (list, tuple)):
+                value_type = type(value)
+                converted_value = []
+                for obj in value:
+                    converted_value.append(obj.to(*unit))
+                converted_value = value_type(converted_value)
+            else:
+                raise ConversionError(f"unable to convert {value}")
+            return converted_value
+
+    class MplUnitsCM:
```

Comment:
bikeshedding, but can this name be changed? Maybe it could be called `unyt_plotting` or `matplotlib_support`. For the second option people can do (and you can write it in the docs as) `with unyt.matplotlib_support`, which I think is easier to remember than the current name. I realize that pep8 says class names should be camel cased but I think in places that classes get exposed in a top-level API like this it’s ok to use lowercase underscore names since they’re easier to type and remember